x/nft/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,583 @@
+			func(require *require.Assertions) {
+				req = &nft.QueryOwnerRequest{
+					ClassId: testClassID,
+				}
+			},
+			"nft id can not be empty",
+			func(require *require.Assertions, res *nft.QueryOwnerResponse) {},
+		},
+		{
+			"success but nft id not exist",
+			func(require *require.Assertions) {
+				req = &nft.QueryOwnerRequest{
+					ClassId: testClassID,
+					Id:      "kitty2",
+				}
+			},
+			"",
+			func(require *require.Assertions, res *nft.QueryOwnerResponse) {
+				require.Equal(res.Owner, owner)
+			},
+		},
+		{
+			"success but class id not exist",
+			func(require *require.Assertions) {
+				req = &nft.QueryOwnerRequest{
+					ClassId: "kitty1",
+					Id:      testID,
+				}
+			},
+			"",
+			func(require *require.Assertions, res *nft.QueryOwnerResponse) {
+				require.Equal(res.Owner, owner)
+			},
+		},
+		{
+			"Success",
+			func(require *require.Assertions) {
+				suite.TestMint()
+				req = &nft.QueryOwnerRequest{
+					ClassId: testClassID,
+					Id:      testID,
+				}
+				owner = suite.addrs[0].String()
+			},
+			"",
+			func(require *require.Assertions, res *nft.QueryOwnerResponse) {
+				require.Equal(res.Owner, owner)
+			},
+		},
+	}
+	for _, tc := range testCases {
+		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
+			require := suite.Require()
+			tc.malleate(require)
+			result, err := suite.queryClient.Owner(gocontext.Background(), req)
+			if tc.expError == "" {
+				require.NoError(err)
+			} else {
+				require.Error(err)
+				require.Contains(err.Error(), tc.expError)
+			}
+			tc.postTest(require, result)
+		})
+	}
+}
+
+func (suite *TestSuite) TestGRPCQuerySupply() {
+	var (
+		req    *nft.QuerySupplyRequest
+		supply uint64
+	)
+	testCases := []struct {
+		msg      string
+		malleate func(require *require.Assertions)
+		expError string
+		postTest func(require *require.Assertions, res *nft.QuerySupplyResponse)
+	}{
+		{
+			"fail empty ClassId",
+			func(require *require.Assertions) {
+				req = &nft.QuerySupplyRequest{}
+			},
+			"class id can not be empty",
+			func(require *require.Assertions, res *nft.QuerySupplyResponse) {},
+		},
+		{
+			"success but class id not exist",
+			func(require *require.Assertions) {
+				req = &nft.QuerySupplyRequest{
+					ClassId: "kitty1",
+				}
+			},
+			"",
+			func(require *require.Assertions, res *nft.QuerySupplyResponse) {
+				require.Equal(res.Amount, supply)
+			},
+		},
+		{
+			"success but supply equal zero",
+			func(require *require.Assertions) {
+				req = &nft.QuerySupplyRequest{
+					ClassId: testClassID,
+				}
+				suite.TestSaveClass()
+			},
+			"",
+			func(require *require.Assertions, res *nft.QuerySupplyResponse) {
+				require.Equal(res.Amount, supply)
+			},
+		},
+		{
+			"Success",
+			func(require *require.Assertions) {
+				n := nft.NFT{
+					ClassId: testClassID,
+					Id:      testID,
+					Uri:     testURI,
+				}
+				err := suite.app.NFTKeeper.Mint(suite.ctx, n, suite.addrs[0])
+				suite.Require().NoError(err)
```

Comment:
why not passing `require *require.Asssertions` as an argument to the function as you do in other tests?